### ISSUE TITLE
Landing pages a11y issues

### ISSUE THREAD

**Issue by olitharp-nci** [NCIOCPL/cgov-digital-platform#3866]
- [ ] Main content areas should always use `<main>` so it has the correct role. 
    ```diff
    - <div class="main-content type {{ page_type }}" id="main-content">...</div>
    + <main class="main-content type {{ page_type }}" id="main-content">...</main>
    
- [ ] Sections should always have a `labelledby={{ some id }}` attribute so we always know a section is used correctly. If there is no label or header, `div` should be used.
    ```diff
    - <section class="usa-section">...</section>
    + <section class="usa-section" labelledby="featureCardRow1">...</section>
    ```
    

Should go ahead and make sure other templates are also correct, but this ticket targets release 2 components, e.g. promo blocks, feature cards, guide cards.
